src/subcommand.go

```diff
@@ -80,222 +144,178 @@ func parseArgs(args []string, cmd *flag.FlagSet) ([]string, []string, error) {
-
-		flags, nonFlags, err := parseArgs(args, cmd)
-		if err != nil {
-			printCmdErr(fmt.Sprintf("Error: %v", err))
-			return
-		}
-
-		reorderedArgs := append(flags, nonFlags...)
-
-		if len(nonFlags) < 1 {
-			printCmdErr("Add subcommand requires at least one input arg append to index log")
-			return
-		}
-
-		cmd.Parse(reorderedArgs)
+}
 
-		if *sourceType == "" {
-			sourceTypeStr, err := inferSourceType(cmd.Args())
-			if err != nil {
-				printCmdErr(fmt.Sprintf("Failed to infer source type from URIs: %v\n", err))
-			}
-			*sourceType = sourceTypeStr
-		}
+func executeAdd(args []string) {
+	cmd := flag.NewFlagSet("add", flag.ExitOnError)
+	dataType := cmd.String("type", "text", "The type of the input data")
+	sourceType := cmd.String("source-type", "", "How to access the content")
+	makeLocalCopy := cmd.Bool("copy", false, "Whether to copy and use the file as it is now, or dynamically access it")
+	indexPath := cmd.String("index-path", "", "Path to the index file")
 
-		args := subcommands.AddArgs{
-			IndexPath:  *indexPath,
-			DataType:   *dataType,
-			SourceType: *sourceType,
-			MakeCopy:   *makeLocalCopy,
-			DataURIs:   cmd.Args(),
-		}
+	flags, nonFlags, err := parseArgs(args, cmd)
+	if err != nil {
+		printCmdErr(fmt.Sprintf("Error: %v", err))
+		return
+	}
 
-		subcommands.Add(args)
+	reorderedArgs := append(flags, nonFlags...)
+	cmd.Parse(reorderedArgs)
 
-	case "delete":
-		cmd := flag.NewFlagSet("delete", flag.ExitOnError)
-		deleteLocalCopy := cmd.Bool("copy", false, "Whether to delete the copy made")
-		indexPath := cmd.String("index-path", "", "Path to the index file")
+	if len(nonFlags) < 1 {
+		printCmdErr("Add subcommand requires at least one input arg append to index log")
+		return
+	}
 
-		flags, nonFlags, err := parseArgs(args, cmd)
+	if *sourceType == "" {
+		sourceTypeStr, err := inferSourceType(cmd.Args())
 		if err != nil {
-			printCmdErr(fmt.Sprintf("Error: %v", err))
-			return
-		}
-
-		reorderedArgs := append(flags, nonFlags...)
-
-		if len(nonFlags) < 1 {
-			printCmdErr("Delete subcommand requires at least one input arg.")
-			return
-		}
-
-		cmd.Parse(reorderedArgs)
-
-		args := subcommands.DeleteArgs{
-			IndexPath:  *indexPath,
-			DeleteCopy: *deleteLocalCopy,
-			DataURIs:   cmd.Args(),
+			printCmdErr(fmt.Sprintf("Failed to infer source type from URIs: %v\n", err))
 		}
+		*sourceType = sourceTypeStr
+	}
 
-		subcommands.Delete(args)
+	addArgs := subcommands.AddArgs{
+		IndexPath:  *indexPath,
+		DataType:   *dataType,
+		SourceType: *sourceType,
+		MakeCopy:   *makeLocalCopy,
+		DataURIs:   cmd.Args(),
+	}
 
-	case "get":
-		cmd := flag.NewFlagSet("get", flag.ExitOnError)
-		indexPath := cmd.String("index-path", "", "Path to the index file")
+	subcommands.Add(addArgs)
+}
 
-		flags, nonFlags, err := parseArgs(args, cmd)
-		if err != nil {
-			printCmdErr(fmt.Sprintf("Error: %v", err))
-			return
-		}
+func executeDelete(args []string) {
```

Comment:
this is ok for now but I think these should live in their respective subcommand files